CHANGELOG.md

```diff
@@ -1,5 +1,78 @@
 # Changelog
 
+## 1.10.0
+
+- ### Features
+
+  - **`wrangler dev` now requires that you specify "preview" versions of KV namespaces - [EverlastingBugstopper], [issue/1032] [pull/1357] [pull/1360]**
+
+    We're preparing to make `wrangler dev` run entirely on Cloudflare's edge, giving it access to functionality that's currently missing in the preview environment, and more accurately reproducing what happens when your Worker is deployed to production. This means that, among other things, your Workers can access the Cache API and your production KV Namespaces.
+
+    In order to prevent you from accidentally stomping on production data in Workers KV or the Cache API, we're introducing the concept of explicit *preview namespaces*. When running `wrangler dev`, if you're using Workers KV, you'll need to specify a specific KV Namespace to use when previewing the Worker.
+
+    Specifically, this change:
+    * Adds a `preview_id` field to items in `kv_namespaces` in `wrangler.toml` that _must_ be provided in order to preview a worker that has kv namespaces.
+    * also adds `--preview` to kv commands in order to interact with them instead of production namespaces.
+
+    If you define a KV Namespace in your `wrangler.toml` but don't specify a `preview_id`, and then try to run `wrangler dev`, you'll see the following:
+    ```console
+    $ wrangler preview
+    💁  JavaScript project found. Skipping unnecessary build!
+    Error: In order to preview a worker with KV namespaces, you must designate a preview_id for each KV namespace you'd like to preview.
+    ```
+
+    More details can be found in the documentation.
+
+    [EverlastingBugstopper]: https://github.com/EverlastingBugstopper
+    [pull/1357]: https://github.com/cloudflare/wrangler/pull/1357
+    [pull/1360]: https://github.com/cloudflare/wrangler/pull/1360
+    [issue/1032]: https://github.com/cloudflare/wrangler/issues/1032
+
+  - **`wrangler subdomain` allows you to rename your existing workers.dev subdomain - [nprogers], [issue/1279] [pull/1353]**
+
+    Workers recently introduced the ability to rename your workers.dev subdomain in the Workers dashboard. This change allows you to do the same thing from `wrangler`.
+
+    [nprogers]: https://github.com/nprogers
+    [pull/1353]: https://github.com/cloudflare/wrangler/pull/1353
+    [issue/1279]: https://github.com/cloudflare/wrangler/issues/1279
+
+  - **Only update wasmpack and cargo-generate when there's a new version - [jspspike], [issue/1240] [pull/1358]**
```

Comment:
```suggestion
  - **Only update wasm-pack and cargo-generate when there's a new version - [jspspike], [issue/1240] [pull/1358]**
```

this one should probably go in maintenance section, and it should probably look like

```markdown
**Pin wasm-pack and cargo-generate to specific versions instead of always downloading the latest version - [jspspike], [issue/1240] [pull/1358]**

Previously, every time `wasm-pack` or `cargo-generate` was needed, Wrangler would download and install the latest version. Now, we pin those dependencies to a specific version in every release of Wrangler. If you have either of these tools installed locally, Wrangler will use them if you have a more recent version.
```

Reply:
Done.